testing_and_setup/compass/ocean/global_ocean/patch/init/cull_mesh_patch.py

```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/env python
+"""
+This script culls all cells except for a lat/lon patch
+"""
+
+from __future__ import absolute_import, division, print_function, \
+    unicode_literals
+
+import os
+import os.path
+import subprocess
+import numpy
+#from optparse import OptionParser
+import xarray
+
+from mpas_tools import conversion
+from mpas_tools.io import write_netcdf
+
+# may add input later
+# parser = OptionParser()
+# parser.add_option("-p", "--geom_data_path", type="string", dest="path",
+#                   default="geometric_data",
+#                   help="Path to the geometric_data from the geometric_features"
+#                        " repository.")
+# options, args = parser.parse_args()
+
+minlat = numpy.deg2rad(30.)
+maxlat = numpy.deg2rad(40.)
+minlon = numpy.deg2rad(30.)
+maxlon = numpy.deg2rad(40.)
+
+# required for compatibility with MPAS
+netcdfFormat = 'NETCDF3_64BIT'
+
+# Create the land mask based on the land coverage, i.e. coastline data.
+dsBaseMesh = xarray.open_dataset('base_mesh.nc')
+
+# These are like pointers -- no copies are made
+# each is an xarray.DataArray
+lat = dsBaseMesh.latCell
+lon = dsBaseMesh.lonCell
+
+# This is a bool type xarray.DataArray
+landMask = numpy.logical_and(numpy.logical_and(lat >= minlat, lat <= maxlat),
+                             numpy.logical_and(lon >= minlon, lon <= maxlon))
+
+# Convert logical to int
+landMask = landMask.astype(int)
+
+dsLandMask = xarray.Dataset()
+dsLandMask['cullCell'] = landMask
+
+dsCulledMesh = conversion.cull(dsBaseMesh, dsMask=dsLandMask,
```

Comment:
@xylar, when I run this case: `-o ocean -c global_ocean -r patch -t init` it runs through successfully and creates `culled_mesh.nc`, but no cells are culled. Perhaps you can see why. Is this dsMask argument looking for an array called `cullCell` within the Dataset called dsLandMask? I'm not very familiar with xarray.